packages/scripts/utils/cli.js

```diff
@@ -21,7 +21,8 @@ const getArgFromCLI = ( arg ) => {
 
 const hasArgInCLI = ( arg ) => getArgFromCLI( arg ) !== undefined;
 
-const getFileArgsFromCLI = () => minimist( getArgsFromCLI() )._;
+const getFileArgsFromCLI = () =>
```

Comment:
I'm a bit worried that it won't work for every tool. Maybe, we should make it an option you pass so we don't break anything accidentally?

Example:
https://github.com/WordPress/gutenberg/blob/1a33c0d87b4da3898216f6d7a300e4af88c6d6ca/package.json#L287

Here it would detect the path to the config as a file passed to the CLI.

I'm afraid that every tool has something like that. I took a look at `stylelint` docs, and they also allow some liberty on how to define options that take values:

https://stylelint.io/user-guide/usage/cli

Example:

`stylelint "foo/**/*.scss" --cache --cache-location "/Users/user/.stylelintcache/"`

The trick they do is placing the list of files immediately after the name of the tool.

Reply:
> I'm a bit worried that it won't work for every tool. Maybe, we should make it an option you pass so we don't break anything accidentally?

Indeed it is.
How about providing a default option as follows?

```javascript
const getFileArgsFromCLI = ( opts = {} ) => minimist( getArgsFromCLI(), opts )._;
or...
const getFileArgsFromCLI = ( useBoolOpt = false ) =>
	minimist( getArgsFromCLI(), useBoolOpt ? { boolean: true } : {} )._;
```

> It's really difficult to tell if there is any good solution here. It looks like the boolean flag just before the list of files is the only issue.

Would it work to explicitly handle `--fix` parameters and target files and pass them to the command in the correct order? 🤔
For example, `wp-scripts format` seems to explicitly control parameters and order as follows:

https://github.com/WordPress/gutenberg/blob/1a33c0d87b4da3898216f6d7a300e4af88c6d6ca/packages/scripts/scripts/format.js#L95-L114

Reply:
> For example, wp-scripts format seems to explicitly control parameters and order as follows:

I don't think the order was explicitly picked because of the issue we discuss. 

We try to parse the command to guess with `minimist` whether the list of files is provided. We use that so people don't have to provide the default search patterns that differ between tools. In general, it works great, but we have this edge case where `minimist` isn't smart enough to recognise whether the boolean param like `--fix` should be followed with the value or not. This is also why the hack I recommended with using `--fix=` resolves the issue.

Reply:
Thank you, it is a more complicated issue than I had thought.
Regarding the solution of this issue, would it be better to wait until the native parser of Node 18 become stable and usable, [as you commented](https://github.com/WordPress/gutenberg/issues/30466#issuecomment-1190318205)?

If so, I would like to close this PR once and for all 👍

Reply:
I guess we better leave it as is, so we don't introduce other bugs. We can keep the issue open until someone finds a magic solution. In the meantime, I better remove `Good First Issue` label as it obviously isn't a good fit anymore 😄

Reply:
Thank you so much for trying this approach. It looked so promising at first check!